pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"bytes"
+	"context"
+	crand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	awsconfig "github.com/kanisterio/kanister/pkg/aws"
+	"github.com/kanisterio/kanister/pkg/controllers/repositoryserver"
+	"github.com/kanisterio/kanister/pkg/format"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kube"
+)
+
+const (
+	repositoryServerTestNamespace    = "repository-server-test-namespace-"
+	repositoryServerTestPod          = "repository-server-test-pod-"
+	repositoryServerTestService      = "repository-server-test-service-"
+	kanisterToolsImage               = "ghcr.io/kanisterio/kanister-tools:0.93.0"
+	kanisterToolsImageEnvName        = "KANISTER_TOOLS"
+	testAwsAccessKeyId               = "AKIAIOSFODNN7EXAMPLE"
+	testAwsAccessSecretKey           = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+	testAwsRegion                    = "us-west-2"
+	testAwsLocationEndpoint          = "http://minio.minio.svc.cluster.local:9000"
+	testAwsS3BucketName              = "tests.kanister.io"
```

Comment:
which suite these tests are going to be run- integration or unit tests?

Reply:
Unit Tests